src/utils.py

```diff
@@ -79,3 +80,56 @@ def chain_id(self) -> int:
         Aligned with https://chainlist.org/
         """
         return {Network.MAINNET: 1, Network.GNOSIS: 100}[self]
+
+
+class LoweredString(fields.String):
+    """Custom marshmallow String field for lowered string"""
+
+    def _deserialize(self, value, *args, **kwargs):
+        if hasattr(value, "lower"):
+            value = value.lower()
+        return super()._deserialize(value, *args, **kwargs)
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        if value is None:
+            return ""
+        return str(value).lower()
+
+
+class TokenSchema(Schema):
+    """TokenSchema CoinSchema for serializing/deserializing token data"""
+
+    address = LoweredString(required=True)
+    decimals = fields.Int(required=True)
+    popularity = fields.Int()
+    symbol = fields.String()
+
+
+class CoinSchema(Schema):
+    """CoinSchema for serializing/deserializing coin data"""
+
+    id = fields.String(required=True)
+    name = fields.String()
+    symbol = fields.String(required=True)
+    rank = fields.Int()
+    is_new = fields.Bool()
+    is_active = fields.Bool()
+    type = fields.String()
+    address = LoweredString(required=True)
+
+
+class CoinsSchema(fields.Dict):
+    """CoinsSchema for containing multiple coinschema-s"""
+
+    @staticmethod
+    def _get_obj(obj, _attr, _default):
+        """Accessor for the dump method"""
+        return obj
+
+    def dump(self, obj: Any):
+        """Serializes data"""
+        return self.serialize("", obj, accessor=self._get_obj)
+
+    def load(self, data: Mapping[str, Any]):
```

Comment:
I believe `Mapping` is equivalent to `dict` (a primitive type) - at least for python >= 3.9. This is not a suggestion, but could save on a (potentially) unnecessary import. Could also be that load is an override of an existing abstract method that we have to match.

Reply:
Ah, thanks for the insight. I didn't know that. I'll try to replace it.